### PR TITLE
[Feat](nereids)when dealing insert into stmt with empty table source, fe returns directly

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
When a LogicalOlapScan has no partitions, transform it to a LogicalEmptyRelation. 
When dealing insert into stmt with empty table source, fe returns directly.

### CODE REVIEW

File: regression-test/suites/load_p0/insert/insert_select_empty_table.groovy [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    sql """
+        create table insert_select_empty_table1(pk int, a int, b int) distributed by hash(pk) buckets 10
+        properties('replication_num' = '1'); 
+        """
+    sql """
+         DROP TABLE IF EXISTS insert_select_empty_table2
+        """
+
+    sql """
+        create table insert_select_empty_table2(pk int, a int, b int) distributed by hash(pk) buckets 10
```
Comment (gavinchou) [inline regression-test/suites/load_p0/insert/insert_select_empty_table.groovy:34]: consider adding more cases, such as  join, union and etc.

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run feut

**Comment by englefly** [comment]
是不是可以直接扩展 EliminateEmptyRelation rule? 所有node直接裁剪了

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateMTMVInfo.java [hunk 2]
```diff
@@ -88,7 +88,7 @@
  */
 public class CreateMTMVInfo {
     public static final Logger LOG = LogManager.getLogger(CreateMTMVInfo.class);
-    public static final String MTMV_PLANER_DISABLE_RULES = "OLAP_SCAN_PARTITION_PRUNE";
+    public static final String MTMV_PLANER_DISABLE_RULES = "OLAP_SCAN_PARTITION_PRUNE,PRUNE_EMPTY_PARTITION";
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateMTMVInfo.java:91]: @seawinde disable this rule will lead to mv have more partition than query?

Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateMTMVInfo.java:91]: the mv plan used by query rewrite is generated by `org.apache.doris.mtmv.MTMVCache#from` which runs the same rules as query.
`MTMV_PLANER_DISABLE_RULES` is only used by mv meta check when create mtmv. it seems has no problems

**Review by seawinde — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**